test/validators/size_validator_test.rb

```diff
@@ -239,6 +239,22 @@ def error_message
     end
   end
 
+  class WithContext < ActiveStorageValidations::SizeValidator::Test
+    # validates :size_with_context, size: { less_than: 2.kilobytes }, on: :custom_context
+
+    test 'generates correct error message' do
+      pt = Size::Portfolio.new(title: 'Matisse')
+      pt.size_with_context.attach(file_10ko)
+
+      refute pt.invalid?
+      pt.valid?(:custom_context)
+
+      assert_equal(
+        ['Size with context file size must be less than 2 KB (current size is 10 KB)'],
+        pt.errors.full_messages
+      )
+    end
+  end
 end
 
```

Comment:
@jozzi05 Could you also add a test with a `:if` option? :)

Reply:
https://github.com/igorkasyanchuk/active_storage_validations/pull/209/commits/90b24e35dbffb0d97d3290ad66680c2bccd154ca